modules/custom/az_paragraphs/az_paragraphs.module

```diff
@@ -34,4 +34,22 @@ function az_paragraphs_theme_registry_alter(&$theme_registry) {
       $theme_registry['field_multiple_value_form']['preprocess functions'][] = 'paragraphs_preprocess_field_multiple_value_form';
     }
   }
+
+  $module_path = drupal_get_path('module', 'az_paragraphs');
+  // Find all .html.twig files in this module's folder recursively.
+  $template_file_objects = drupal_find_theme_templates($theme_registry, '.html.twig', $module_path);
+  // Iterate through all found template file objects.
+  foreach ($template_file_objects as $key => $template_file_object) {
+    // If the template has not already been overridden by a theme.
+    if (!isset($theme_registry[$key]['theme path']) || !preg_match('#/themes/#', $theme_registry[$key]['theme path'])) {
+      // Copy default paragraph registry entry for paragraph entries.
+      if (strpos($key, 'paragraph__') === 0) {
+        $theme_registry[$key] = $theme_registry['paragraph'];
+      }
```

Comment:
Most of the code here I borrowed from our D7 uaqs_content_chunks module but I found I neeeded to add this bit because without it, the resulting `$theme_registry` entry for `paragraph__az_text_media__default` was missing the standard set of paragraph preprocess functions.

Debugging output from `$theme_registry` before lines above added:
```
[paragraph__az_text_media__default] => Array
    (
        [theme path] => profiles/custom/az_quickstart/modules/custom/az_paragraphs
        [template] => paragraph--az-text-media--default
        [path] => profiles/custom/az_quickstart/modules/custom/az_paragraphs/az_paragraphs_text_media/templates
        [render element] => elements
        [base hook] => paragraph
        [type] => module
    )
```

Debugging output from `$theme_registry` after lines above added:
```
[paragraph__az_text_media__default] => Array
    (
        [base hook] => paragraph
        [type] => module
        [theme path] => profiles/custom/az_quickstart/modules/custom/az_paragraphs/az_paragraphs_text_media
        [template] => paragraph--az-text-media--default
        [path] => profiles/custom/az_quickstart/modules/custom/az_paragraphs/az_paragraphs_text_media/templates
        [preprocess functions] => Array
            (
                [0] => template_preprocess
                [1] => template_preprocess_paragraph
                [2] => contextual_preprocess
                [3] => smart_title_preprocess
            )

        [render element] => elements
    )
```

Reply:
I also experimented with implementing `hook_theme()`, which caused the correct template to get used from the module directory but I think doing things this way would require adding a `hook_theme()` entry for every view mode of every paragraph type we want to be able to support a custom template for:
```
/**
 * Implements hook_theme().
 */
function az_paragraphs_text_media_theme() {
  $theme = [];
  $theme['paragraph__az_text_media__default'] = [
    'base hook' => 'paragraph',
  ];
  return $theme;
}
```

Reply:
Maybe https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21theme.api.php/function/hook_theme_suggestions_HOOK_alter/8.2.x

Or am I not understanding this?